### PR TITLE
feat(prediction): add legend

### PR DESCRIPTION
**Pull request by csouchet**
**Live environment of examples**

https://cdn.statically.io/gh/process-analytics/bpmn-visualization-examples/260-add_legend/demo/predictions/index.html

https://github.com/process-analytics/bpmn-visualization-examples/assets/4921914/457f98a1-575d-4b93-8577-6b352967b73c

Depends on #524

Covers #260

### CODE REVIEW

File: demo/predictions/css/legend.css [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    padding: 0 10px;
+    margin: auto;
+}
+
+.state-predicted-on-time .legend li:first-child::before {
+    background-color: var(--color-state-predicted-on-time);
+}
+
+.state-predicted-late .legend li:first-child::before {
+    background-color: var(--color-state-predicted-late);
```
Comment (csouchet) [inline demo/predictions/css/legend.css:17]: **question:**  Why there is another variable with the same value (`--color-path-predicted-late`) ?
It's confusing and it's weird to add the 2 colors on the legend 🙁

File: demo/predictions/css/legend.css [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+legend {
+    font-size: 18px;
+    padding: 0 10px;
+    margin: auto;
+}
+
+.state-predicted-on-time .legend li:first-child::before {
+    background-color: var(--color-state-predicted-on-time);
```
Comment (csouchet) [inline demo/predictions/css/legend.css:13]: **suggestion:** This question is similar to the one about `--color-state-predicted-late`, but in this case, the values for `--color-state-predicted-on-time` and `--color-path-predicted-on-time` are different. 
As a result, I am uncertain about which color to include in the legend, which is quite frustrating. 🙁

Comment (tbouffard) [reply demo/predictions/css/legend.css:13]: I agree, I don't remember why we didn't choose to use the same value, this is weird.
I let you choose the color 😄

File: demo/predictions/css/legend.css (hunk 1 above)
Comment (tbouffard) [reply demo/predictions/css/legend.css:17]: Same as for the other color, go to keep a single color

**Comment by csouchet** [comment]
Validated by #527

**Pull request closed without merging**